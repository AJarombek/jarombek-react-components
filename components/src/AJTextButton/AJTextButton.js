/**
 * Reusable text button component.
 * @author Andrew Jarombek
 * @since 11/16/2019
 */

import React from 'react';

const AJTextButton = () => {
  return (
    <div>Text Button</div>
  );
};

export default AJTextButton;
