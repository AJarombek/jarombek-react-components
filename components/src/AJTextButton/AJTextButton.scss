/**
 * Stylesheet for the AJTextButton component.
 * Author: Andrew Jarombek
 * Date: 11/23/2019
 */

@import "../global";

.aj-text-button {
  @include RobotoThin;
  padding: 10px;
  display: inline-block;
  cursor: pointer;
  transition: background-color 0.25s ease;

  &.aj-text-button-enabled {
    cursor: pointer;

    &:active {
      background-color: #f5f5f5;
    }

    button {
      cursor: pointer;
      color: #555;
      transition: color 0.25s ease;

      &:active {
        color: #222;
      }
    }
  }

  &.aj-text-button-disabled {
    cursor: not-allowed;

    button {
      cursor: not-allowed;
      color: #999;
    }
  }

  button {
    display: block;
    border: none;
    background-color: transparent;

    &:focus {
      outline: none;
    }
  }
}
