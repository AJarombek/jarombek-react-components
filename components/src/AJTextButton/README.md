### Overview

Reusable button component with a specific "text" style.  For a more generic button, see `AJButton`.

**Use cases:**

```javascript
import { AJTextButton } from 'jarombek-react-components';

const CustomComponent = () => 
  <AJTextButton disabled={false}>Text Button</AJTextButton>
```

### Files

| Filename                   | Description                                                                      |
|----------------------------|----------------------------------------------------------------------------------|
| `AJTextButton.js`          | The reusable `AJTextButton` component.                                           |
| `index.js`                 | Index file which exports the `AJTextButton` component.                           |
| `styles.js`                | JSS styles for the `AJTextButton` component.                                     |
