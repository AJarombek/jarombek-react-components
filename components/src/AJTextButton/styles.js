/**
 * Stylesheet for the AJTextButton component using JSS.
 * Author: Andrew Jarombek
 * Date: 12/30/2019
 */

import globalStyles from '../globalStyles';

export default {
  ajTextButton: {
    padding: '10px',
    transition: 'background-color 0.25s ease',

    '& button': {
      ...globalStyles.robotoRegular,
      display: 'block',
      backgroundColor: 'transparent'
    }
  },
  ajTextButtonEnabled: {
    '&:active': {
      backgroundColor: '#f5f5f5'
    },

    '& button': {
      color: '#555',
      transition: 'color 0.25s ease',

      '&:active': {
        color: '#222'
      }
    }
  },
  ajTextButtonDisabled: {
    '& button': {
      color: '#999'
    }
  }
};
