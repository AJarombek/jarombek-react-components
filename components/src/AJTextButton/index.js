import AJTextButton from 'AJTextButton';

export default AJTextButton;
