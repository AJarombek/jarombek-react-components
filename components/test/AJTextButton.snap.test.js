import React from 'react';
import AJTextButton from '../src/AJTextButton/AJTextButton';
import renderer from 'react-test-renderer';

/**
 * Snapshot test for the {@link ../../components/AJTextButton} component.
 * @author Andrew Jarombek
 * @since 12/19/2019
 */

it('renders correctly', () => {
  const tree = renderer.create(<AJTextButton children={<div>Snapshot Test</div>}/>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with all props', () => {
  const tree = renderer.create(
    <AJTextButton
      children={<div>Snapshot Test</div>}
      onClick={f=>f}
      disabled={true}/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
