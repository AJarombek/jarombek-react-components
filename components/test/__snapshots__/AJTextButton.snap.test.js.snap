// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="aj-text-button aj-text-button-enabled"
>
  <button
    disabled={false}
    type="button"
  >
    <div>
      Snapshot Test
    </div>
  </button>
</div>
`;

exports[`renders correctly with all props 1`] = `
<div
  className="aj-text-button aj-text-button-disabled"
  onClick={null}
>
  <button
    disabled={true}
    type="button"
  >
    <div>
      Snapshot Test
    </div>
  </button>
</div>
`;
